### ISSUE TITLE
Type error after upgraded to nim v0.20.0

### ISSUE THREAD

**Issue by Stii** [Vladar4/nimgame2#30]
After upgrading nim to v0.20.0 got the following when compiling:

```bash
Hint: used config file '/Users/stii/.choosenim/toolchains/nim-0.20.0/config/nim.cfg' [Conf]           
Hint: system [Processing]                          
Hint: widestrs [Processing]                        
Hint: io [Processing]                              
Hint: bounce [Processing]                          
Hint: nimgame [Processing]                         
Hint: math [Processing]                            
Hint: bitops [Processing]                          
Hint: macros [Processing]                          
Hint: parseutils [Processing]                      
Hint: random [Processing]                          
Hint: algorithm [Processing]                       
Hint: times [Processing]                           
Hint: strutils [Processing]                        
Hint: unicode [Processing]                         
Hint: options [Processing]                         
Hint: typetraits [Processing]                      
Hint: posix [Processing]                           
Hint: sdl [Processing]                             
Hint: sdl_image [Processing]                       
Hint: sdl_ttf [Processing]                         
Hint: sdl_mixer [Processing]                       
Hint: audio [Processing]                           
Hint: types [Processing]                           
Hint: utils [Processing]                           
Hint: texturegraphic [Processing]                  
Hint: graphic [Processing]                         
Hint: settings [Processing]                        
Hint: parsecsv [Processing]                        
Hint: lexbase [Processing]                         
Hint: streams [Processing]                         
Hint: indexedimage [Processing]                    
/Users/stii/.nimble/pkgs/nimgame2-0.6.0/nimgame2/audio.nim(221, 21) Error: type mismatch: got <Channel, Distance>                                                                                           
but expected one of:                               
proc setDistance(channel: cint; distance: uint8): cint                                                
  first type mismatch at position: 2               
  required type: uint8                             
  but expression 'distance' is of type: Distance   

expression: mix.setDistance(channel, distance) 
```

It used to compile perfectly with nim v0.19.6, but I think with [the stricter compile time checks for integer conversions](https://nim-lang.org/blog/2019/06/06/version-0200-released.html) broke it.

**Comment by Stii** [comment]
In fact, I think there is something else wrong. Sorry about that.

When I look at the files that `nimble install nimgame2` extracted it had line 33 as:

```nim
type
  Channel* = int
  Distance* = range[0..255]
```

but when I look at the source here it is defined correctly as:

```nim
type
  Channel* = int
  Distance* = range[0'u8..255'u8]
```

Will dig deeper as to how this happened. I think the stricter type conversions might make the version nimble installed break.

**Comment by Stii** [comment]
Got it. The [release tarball](https://github.com/Vladar4/nimgame2/archive/v0.6.tar.gz) has the `0'u8..255'u8` converted to `0..255` which isn't valid anymore. Not sure if you used something like `nimpretty` to format the code before creating a release and if that possibly changed it? 

Anyway, a workaround is to clone the repo rather than use nimble to install.

**Comment by SolitudeSF** [comment]
`nimble install` installs latest tag if present. the fixed version just wasnt tagged.

you can `nimble install 'nimgame2@#HEAD'` to install `HEAD` with nimble.

**Comment by Vladar4** [comment]
Should be fixed with v0.6.1 (along with other adjustments for Nim v0.20.0).
